src/conversion.jl

```diff
@@ -40,33 +41,70 @@ GI.getfeature(::Any, d::Data, i) = d[i,:]
 # Convert geometries to Meshes.jl types
 # --------------------------------------
 
-coords2point(coords) = Point(coords)
-
-coords2multipoint(coords) = Multi(coords2point.(coords))
+function getpoints(geom, is3d::Bool)
+  if is3d
+    [Point(GI.x(p), GI.y(p), GI.z(p)) for p in GI.getpoint(geom)]
+  else
+    [Point(GI.x(p), GI.y(p)) for p in GI.getpoint(geom)]
+  end
+end
 
-coords2chain(coords) = Chain(Point.(coords))
+function GI.convert(::Type{Point}, ::GI.PointTrait, geom)
+  if GI.is3d(geom)
+    return Point(GI.x(geom), GI.y(geom), GI.z(geom))
+  else
+    return Point(GI.x(geom), GI.y(geom))
+  end
+end
 
-coords2multichain(coords) = Multi(coords2chain.(coords))
+function GI.convert(::Type{Segment}, ::GI.LineTrait, geom)
+  p1, p2 = GI.startpoint(geom), GI.endpoint(geom)
+  if GI.is3d(geom)
+    Segment(Point(GI.x(p1), GI.y(p1), GI.z(p1)), Point(GI.x(p2), GI.y(p2), GI.z(p1)))
+  else
+    Segment(Point(GI.x(p1), GI.y(p1)), Point(GI.x(p2), GI.y(p2)))
+  end
```

Comment:
Maybe we could use `Segment(getpoints(geom))` here as well?

Reply:
Yes, that would also works and make it cleaner.